File: Backend/src/main/java/com/bachld/project/backend/dto/response/hoidong/HoiDongDetailResponse.java
package com.bachld.project.backend.dto.response.hoidong;

import lombok.*;
import lombok.experimental.FieldDefaults;
import java.time.LocalDate;
import java.util.List;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class HoiDongDetailResponse {

    Long id;
    String tenHoiDong;
    LocalDate thoiGianBatDau;
    LocalDate thoiGianKetThuc;
    String loaiHoiDong;

    String chuTich;
    String thuKy;
    List<String> giangVienPhanBien;

    List<SinhVienTrongHoiDong> sinhVienList;

    @Builder
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @FieldDefaults(level = AccessLevel.PRIVATE)
    public static class SinhVienTrongHoiDong {
        String hoTen;
        String maSV;
        String lop;
        String tenDeTai;
        String gvhd;
        String boMon;
    }
}

File: Android/app/src/main/java/com/bachld/android/data/dto/response/hoidong/HoiDongDetailResponse.kt
package com.bachld.android.data.dto.response.hoidong

data class HoiDongDetailResponse(
    val id: Long,
    val tenHoiDong: String,
    val thoiGianBatDau: String,
    val thoiGianKetThuc: String,
    val loaiHoiDong: String?,
    val chuTich: String?,
    val thuKy: String?,
    val giangVienPhanBien: List<String> = emptyList(),
    val sinhVienList: List<SinhVienTrongHoiDong> = emptyList()
) {
    data class SinhVienTrongHoiDong(
        val hoTen: String,
        val maSV: String,
        val lop: String,
        val tenDeTai: String,
        val gvhd: String,
        val boMon: String
    )
}
